### PR TITLE
gh-113317: Move global utility functions into libclinic

### PR DESCRIPTION
**Pull request by erlend-aasland**
Establish Tools/clinic/libclinic/utils.py and move the following
functions over there:

- compute_checksum()
- create_regex()
- write_file()

* Issue: gh-113317

### CODE REVIEW

File: Tools/clinic/libclinic/utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def create_regex(
+    before: str, after: str, word: bool = True, whole_line: bool = True
+) -> re.Pattern[str]:
+    """Create a regex object for matching marker lines."""
+    group_re = r"\w+" if word else ".+"
+    pattern = r"{}({}){}"
+    if whole_line:
+        pattern = "^" + pattern + "$"
+    pattern = pattern.format(re.escape(before), group_re, re.escape(after))
```
Comment (AA-Turner) [inline Tools/clinic/libclinic/utils.py]: Perhaps worth doing if you're moving the function anyway?
```suggestion
    pattern = rf"{re.escape(before)}({group_re}){re.escape(after)}"
    if whole_line:
        pattern = f"^{pattern}$"
```

**Review by AA-Turner — APPROVED** [review]

File: Tools/clinic/libclinic/utils.py (hunk 1 above)
Comment (erlend-aasland) [reply Tools/clinic/libclinic/utils.py]: Thanks, but I think there is too much happening in the resulting line 40. It is too easy to not see the overall structure of the regex (`{pre}({group}){post}`). How about this:

```suggestion
    pattern = r"{before}({group}){after}"
    if whole_line:
        pattern = fr"^{pattern}$"
    pattern = pattern.format(
        before=re.escape(before),
        group=group_re,
        after=re.escape(after)
    )
```

Alternatively:

```suggestion
    before = re.escape(before)
    after = re.escape(after)
    pattern = fr"{before}({group_re}){after}"
    if whole_line:
        pattern = fr"^{pattern}$"
```

Comment (AlexWaygood) [reply Tools/clinic/libclinic/utils.py]: Drive-by comment: FWIW, I quite like Adam's suggestion, but I also like Erlend's second suggestion. Either would be fine to me :)

Comment (erlend-aasland) [reply Tools/clinic/libclinic/utils.py]: Thanks, both! I applied my second suggestion ;)

**Pull request merged**